File: js/weightedStrings.test.js
const weightedString = require('./weightedStrings')

test('weightedString with input "abbcccd" and [1, 3, 9, 8] to equal ["YES", "YES", "YES", "NO"]', () => {
    expect(weightedString("abbcccd", [1, 3, 9, 8])).toEqual(["YES", "YES", "YES", "NO"]);
});

test('weightedString with input "aaabbbcccddd" and [5, 9, 7, 8, 12, 5] to equal ["NO", "YES", "NO", "YES", "YES", "NO"]', () => {
    expect(weightedString("aaabbbcccddd", [5, 9, 7, 8, 12, 5])).toEqual(["NO", "YES", "NO", "YES", "YES", "NO"]);
});

test('weightedString with input "getitssimple" and [1, 2, 8, 7, 12, 38, 10] to equal [NO, NO, NO, YES, YES, YES, NO]', () => {
    expect(weightedString("getitssimple", [1, 2, 8, 7, 12, 38, 10])).toEqual(["NO", "NO", "NO", "YES", "YES", "YES", "NO"]);
});

File: java-test/app/src/test/java/talent/assesment/smkdev/WeightedStringsTest.java
package talent.assesment.smkdev;

import org.junit.Test;
import static org.junit.Assert.*;
import java.util.List;

public class WeightedStringsTest {

    @Test
    public void testWeightedStringsCase1() {
        List<String> expected = List.of("Yes", "Yes", "Yes", "No");
        assertEquals(expected, WeightedStrings.weightedStrings("abbcccd", new int[]{1, 3, 9, 8}));
    }

    @Test
    public void testWeightedStringsCase2() {
        List<String> expected = List.of("No", "Yes", "No", "Yes", "Yes", "No");
        assertEquals(expected, WeightedStrings.weightedStrings("aaabbbcccddd", new int[]{5, 9, 7, 8, 12, 5}));
    }

    @Test
    public void testWeightedStringsCase3() {
        List<String> expected = List.of("No", "No", "No", "Yes", "Yes", "Yes", "No");
        assertEquals(expected, WeightedStrings.weightedStrings("getitssimple", new int[]{1, 2, 8, 7, 12, 38, 10}));
    }
}
